### PR TITLE
DRAFT with just an initialize method

### PR DESCRIPTION
**Pull request by apupier**
the telemetry attribute retrieved by the stub is undefined:

https://app.circleci.com/pipelines/github/camel-tooling/vscode-camelk/1065/workflows/107d9edc-abd8-480a-9476-5677ef87c741/jobs/1355?invite=true#step-110-442

### CODE REVIEW

**Comment by fbricon** [comment]
That looks good on paper. Now the test failure seems to indicate the service is not initialized during testing

File: src/extension.ts [hunk 1]
```diff
@@ -66,6 +66,7 @@ const COMMAND_ID_OPEN_OPERATOR_LOG = 'camelk.integrations.openOperatorLog';
 export const COMMAND_ID_START_JAVA_DEBUG = 'camelk.integrations.debug.java';
 export async function activate(context: vscode.ExtensionContext) {
 	stashedContext = context;
+	telemetry.initializeTelemetry(context);
```
Comment (fbricon) [inline src/extension.ts:69]: try to await here?

Comment (apupier) [reply src/extension.ts]: still same error I do not think that it is not a problem of timing. When I tried initially, i put traces and they were written in correct order.

**Comment by apupier** [comment]
> That looks good on paper.

agree. i do not understand why it doesn't work. it sounds like there are different instances although it is a kind of static access

> Now the test failure seems to indicate the service is not initialized during testing

When initially tried, i put traces and it was showing that the initialization was done but the telemetry retrieved by the mock framework is not the one used by the product during test execution.

File: package.json [hunk 2]
```diff
@@ -365,8 +365,7 @@
 	},
 	"extensionDependencies": [
 		"ms-kubernetes-tools.vscode-kubernetes-tools",
-		"redhat.vscode-yaml",
-		"redhat.vscode-commons"
+		"redhat.vscode-yaml"
 	],
 	"dependencies": {
 		"@redhat-developer/vscode-redhat-telemetry": "^0.1.1",
```
Comment (fbricon) [inline package.json:371]: ^0.2.0

Comment (apupier) [reply package.json]: I was trying to keep the change minimal to demonstrate it. But updated and same error https://app.circleci.com/pipelines/github/camel-tooling/vscode-camelk/1069/workflows/06ba6b81-9666-487c-8bf1-3c3997b9b51e/jobs/1359?invite=true#step-110-452

**Comment by fbricon** [comment]
So I can see Telemetry.ts is being instanciated twice, once when the extensions' activate function is called, and another time. That second time, the initialize method is not called, explaining the failures.
So most likely, 2 sets of tests are being run (UI vs other?). Whatever is launching the 2nd type of test should ensure initialize is being called.

File: src/extension.ts [hunk 3]
```diff
@@ -212,7 +214,7 @@ export async function activate(context: vscode.ExtensionContext) {
 	
 	return {
 		getStashedContext() : vscode.ExtensionContext {
-			return stashedContext;
+			return stashedContext;Will initialize Telemetry
```
Comment (fbricon) [inline src/extension.ts:217]: is that a comment?

Comment (apupier) [reply src/extension.ts:217]: that's a bad copy when I accidently click on the mouse scroll wheel. i will remove it

**Comment by apupier** [comment]
siwtched back to VS Code 1.57.1 to continue testing. And wouhou, the mock is better workign when the coverage is deactivated.

So now the question is how to have both of them workign at the same time? Maybe need to switch to nyc as istanbul is deprecated.
or we deactivate the coverage for now to have simpler code. The coverage is using deprecated api anyway.

**Comment by fbricon** [comment]
Maybe call telemetry.initialize explicitly when starting the code coverage (use a mock context if needed)

**Comment by apupier** [comment]
> Maybe call telemetry.initialize explicitly when starting the code coverage (use a mock context if needed)

The problem will be that I won't test anymore that the Telemetry is initialized when the activate method is called. Or need to add another tests checking for that with spies but it seems that we start to have weak and complex tests. Not sure it would be worthy.

I think I would prefer to deactivate coverage for some time and plan a migration to nyc and find a way to have it working properly.

File: src/test/runTests.ts [hunk 4]
```diff
@@ -14,7 +14,8 @@ async function main() : Promise<void> {
 		if(vscodeVersionForTest !== undefined){
 			vscodeExecutablePath = await downloadAndUnzipVSCode(vscodeVersionForTest);
 		} else {
-			vscodeExecutablePath = await downloadAndUnzipVSCode('stable');
+			// FIXME: using previous version to workaround https://github.com/microsoft/vscode/issues/126636
+			vscodeExecutablePath = await downloadAndUnzipVSCode('1.57.1');
 		}
 		console.log(`vscodeExecutablePath = ${vscodeExecutablePath}`);
 		const testWorkspace = path.resolve(__dirname, '../../../test Fixture with speci@l chars');
```
Comment (fbricon) [inline src/test/runTests.ts:21]: need to remove installExtraExtension(cliPath, 'redhat.vscode-commons'); down there

**Comment by apupier** [comment]
created cleaner PRs:
- to remove istanbul: https://github.com/camel-tooling/vscode-camelk/pull/842
- use new API https://github.com/camel-tooling/vscode-camelk/pull/844
- upgrade telemetry package https://github.com/camel-tooling/vscode-camelk/pull/845
@fbricon thanks a lot for the help

**Pull request closed without merging**